### DISCUSSION TITLE
Page transition?

### DISCUSSION THREAD

**Discussion by NeighborhoodNerd** [taublast/DrawnUi#104] [General]
Is there any way to set the page transition speed or maybe have custom transition animations? The demo app seems to be a bit slow compared to what I'm used to with an out of the box MAUI app on iOS.

**Comment by taublast** [comment]
Hi, i guess we are talking about `SkiaViewSwitcher` here, I see that there are currently some hardcoded stuff there:

![image](https://github.com/user-attachments/assets/94ebc076-e239-4c36-9ed2-a6f77f455d30)

so what I could do:
1 virtual methods for animation so one can subclass `SkiaViewSwitcher` and implement custom transitions upon type (Pop, Push etc) inside override.
2 static settings for speed etc to affect unchanged `SkiaViewSwitcher`, for example usable like "SkiaViewSwitcher.PageNavigationSpeed = 100;".

Thoughts?

**Reply by NeighborhoodNerd** [reply]
Well, I think users would like the ability to apply certain transitions to certain pages so could this be done with defining a property on the SkiaLayout control? Maybe take a page out of Avalonia's book? 

https://github.com/AvaloniaUI/Avalonia/blob/master/src/Avalonia.Base/Animation/PageSlide.cs

**Comment by taublast** [comment]
Looks nice as a solid addition direction. Can already tell that it would be inline with the possibility to create shader transitions along with simple animations. 

So if you need a quick fix can do points 1+2 this week as it costs nothing in time, while this more solid option would require "a bit" more time, but will surely do.

**Reply by NeighborhoodNerd** [reply]
Sounds good! I'm trying to study the codebase so I can eventually help out but it's pretty complex 😅
